frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,437 @@
+        disableForm: disable
+      });
+    }
+    // Add Attach storage //
+    save(e) {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.kind === kind);
+      let resourceObj = { ...this.state.resourceObj };
+      let {
+        volumeName,
+        mountPath,
+        subPath,
+        claimName,
+        readOnly
+      } = this.state.stringData;
+
+      var supportedKinds = [
+        "Deployment",
+        "DeploymentConfig",
+        "ReplicaSet",
+        "ReplicationController",
+      ];
+
+      if (!_.includes(supportedKinds, ko.kind)) {
+        err =>
+          this.setState({
+            error: "Storage is not supported for kind " + ko.kind + ".",
+            inProgress: false
+          });
+        return;
+      }
+      let storage = {
+        attach: null,
+        existingMountPaths: null,
+        volumeAlreadyMounted: null,
+        onPVCSelected: null
+      };
+      storage.attach = {
+        persistentVolumeClaim: claimName,
+        volumeName: volumeName,
+        mountPath: mountPath,
+        subPath: subPath,
+        readOnly: readOnly,
+        allContainers: true,
+        containers: {}
+      };
+      var isContainerSelected = function(container) {
+        return (
+          storage.attach.allContainers ||
+          storage.attach.containers[container.name]
+        );
+      };
+
+      //   Look at the existing mount paths so that we can warn if the new value is not unique.
+      storage.existingMountPaths = getMountPaths(
+        _.get(resourceObj.spec, "template"),
+        isContainerSelected
+      );
+      var updateVolumeName = function() {
+        var pvc = _.get(resourceObj.spec, "volumes.persistentVolumeClaim");
+        if (!pvc) {
+          return;
+        }
+
+        // Check if there is already a volume for this PVC.
+        var volumes = _.get(resourceObj.spec, "template.spec.volumes");
+        var volume = _.find(volumes, {
+          persistentVolumeClaim: {
+            claimName: pvc.metadata.name
+          }
+        });
+        if (volume) {
+          // If there's already a  volume, reuse the volume name.
+          storage.attach.volumeName = resourceObj.spec.volume.metadata.name;
+          storage.volumeAlreadyMounted = true;
+        } else if (storage.volumeAlreadyMounted) {
+          // Clear the volume name value since it was associated with the previously selected PVC.
+          storage.attach.volumeName = "";
+          storage.volumeAlreadyMounted = false;
+        }
+      };
+      storage.onPVCSelected = updateVolumeName;
+
+      // generate a volume name if not provided
+      if (!volumeName) {
+        // Create a volume name using the project
+        volumeName = "volume-" + metadata.namespace;
+      }
+      if (mountPath) {
+        // for each container in the pod spec, add the new volume mount
+
+        _.forEach(resourceObj.spec.template.spec.containers, function(
```

Comment:
style nit: prefer the shorter `_.each`

We typically don't put single function arguments on their own line